src/Features/CSharp/Portable/ConvertAnonymousTypeToClass/CSharpConvertAnonymousTypeToClassCodeRefactoringProvider.cs

```diff
@@ -41,7 +42,19 @@ private ArgumentListSyntax CreateArgumentList(AnonymousObjectCreationExpressionS
                 SyntaxFactory.Token(SyntaxKind.CloseParenToken).WithTriviaFrom(anonymousObject.CloseBraceToken));
 
         private SeparatedSyntaxList<ArgumentSyntax> CreateArguments(SeparatedSyntaxList<AnonymousObjectMemberDeclaratorSyntax> initializers)
-            => SyntaxFactory.SeparatedList<ArgumentSyntax>(CreateArguments(initializers.GetWithSeparators()));
+            => SyntaxFactory.SeparatedList<ArgumentSyntax>(CreateArguments(OmitTrailingComma(initializers.GetWithSeparators())));
+
+        private static SyntaxNodeOrTokenList OmitTrailingComma(SyntaxNodeOrTokenList list)
+        {
+            // Trailing comma is allowed in initializer list, but disallowed in method calls.
+            return list.Count >= 2 && list[^1].IsToken
+                ? list
+                    .Replace(
+                        list[^2],
+                        list[^2].AsNode().WithAppendedTrailingTrivia(list[^1].GetTrailingTrivia()))
```

Comment:
❔ Shouldn't this be leading and trailing trivia?

```suggestion
                        list[^2].AsNode().WithAppendedTrailingTrivia(list[^1].GetLeadingTrivia()).WithAppendedTrailingTrivia(list[^1].GetTrailingTrivia()))
```

Reply:
You are right! Fixed.

Reply:
By the way, currently some trivia of deleted parts in the initializer list is lost. For example,
```
            var t1 = new
            {
                // Hi
                a = 1,
                b = 2
            };
```
is transformed into
```
            var t1 = new NewClass(
1,
2
            );
```
Is this considered a bug?

Reply:
yup, i would def consider that a bug :)

Reply:
Preserving leading trivia from the initializers would also produce more reasonable indentation.

Reply:
yes.  i woudl consider preservatin of leading trivia appropriate, as well as the removal of the final trailing trivia of the last element.

Reply:
Created issue #45792